js/engine.js

```diff
@@ -14,7 +14,7 @@
  * a little simpler to work with.
  */
 
-var Engine = (function(global) {
+var Engine = function(global) {
```

Comment:
From what I can tell, the immediately executed function is returning undefined due to the fact that it has no return value. That is being assigned to Engine. This, the way I see it, is causing the Engine variable not to be accessible in other files (which may be the intention, but the in-code comments and other users seem to indicate otherwise). By removing the IIFE, assigning a regular function to Engine and invoking it afterward, the variable holds the proper reference to the function and maintains the same functionality as before, while being accessible from other files.